cli/azd/pkg/auth/manager.go

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package auth
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/public"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+// TODO(azure/azure-dev#710): Right now, we re-use the App Id of the `az` CLI, until we have our own.
+//
+// nolint:lll
+// https://github.com/Azure/azure-cli/blob/azure-cli-2.41.0/src/azure-cli-core/azure/cli/core/auth/identity.py#L23
+const cAZD_CLIENT_ID = "04b07795-8ddb-461a-bbee-02f9e1bf7b46"
+
+// cCurrentUserKey is the key we use in config for the storing identity information of the currently logged in user.
+const cCurrentUserKey = "auth.account.currentUser"
+
+// cUseAzCli is the key we use in config to denote that we want to use the az CLI for authentication instead of managing
+// it ourselves. The value should be a string as specified by [strconv.ParseBool].
+const cUseLegacyAzCliAuthKey = "auth.useLegacyAzCliAuth"
+
+// The scopes to request when acquiring our token during the login flow or when requesting a token to validate if the client
+// is logged in.
+var cLoginScopes = []string{"https://management.azure.com//.default"}
+
+// authDirectoryFileMode is the file mode used to create the folder that is used for auth folder and sub-folders.
+const authDirectoryFileMode = 0700
```

Comment:
Consider moving this into our `osutil` package where we store our other commonly used file & directory permissions.

Reply:
created osutil.PermissionDirectoryOwnerOnly and osutil.PermissionFileOwnerOnly and moved to them.